src/mtxLora/CommandBinaryBuffer.ts

```diff
@@ -0,0 +1,396 @@
+import BinaryBuffer from '../utils/BinaryBuffer.js';
+import * as bitSet from '../utils/bitSet.js';
+
+
+export interface IDate {
+    year: number,
+    month: number,
+    day: number
+}
+
+const energiesMask = {
+    aPlus: 0x01,
+    aPlusRPlus: 0x02,
+    aPlusRMinus: 0x04,
+    aMinus: 0x08,
+    aMinusRPlus: 0x10,
+    aMinusRMinus: 0x20
+};
+
+export interface IEnergies<T = number> {
+    aPlus?: T,
+    aPlusRPlus?: T,
+    aPlusRMinus?: T,
+    aMinus?: T,
+    aMinusRPlus?: T,
+    aMinusRMinus?: T
+}
+
+export type TEnergiesFlags = IEnergies<boolean>;
+
+const getEnergiesFlags = <T>( energies: IEnergies<T> ): number => {
+    const booleanObject = Object.fromEntries(Object.entries(energies).map(([name, value]) => [name, !!value]));
+
+    return bitSet.fromObject(energiesMask, (booleanObject as unknown) as bitSet.TBooleanObject);
+};
+
+type THalfhoursEnergy = Array<number | undefined>;
+
+export type THalfhoursEnergies = IEnergies<THalfhoursEnergy>;
+
+const UndefinedEnergyValue = 0xffffffff;
```

Comment:
```suggestion
const undefinedEnergyValue = 0xffffffff;
```

?
or maybe in UPPER_CASE

Reply:
fixed